### PR TITLE
Prepare for test suite

### PR DESCRIPTION
**Pull request by adreghiciu**
1. Switch to not install plugins if they exist in bundle used by IT
   This switch will allow running ITs against a bundle without installing the plugins specified by IT, in case that those plugins already exist in bundle (compared by groupId / artifactId)
2. Make possible to filter nexus bundle coordinates passed to its
   The new ITs are passing as ctor the coordinates fo teh bundle to run against. Make possible to filter that value using project/system properties

### CODE REVIEW

**Comment by peterlynch** [comment]
+1 looks ok, but I suggest a feature build to be sure.

**Comment by adreghiciu** [comment]
Here it is: https://builds.sonatype.org/view/nexus-features/job/nexus-oss-its-feature/451/

**Comment by adreghiciu** [comment]
CI voted +1

**Comment by jdillon** [comment]
Does this need to be coordinated with plugin changes? I saw some plugins were updated for this stuff...

**Comment by adreghiciu** [comment]
This will not break any existing code. When this is in we can merge the changes to plugins to make use of what we did here.

File: nexus/nexus-launcher/src/main/java/org/sonatype/nexus/bundle/launcher/support/DefaultNexusBundle.java [hunk 1]
```diff
@@ -258,6 +277,122 @@ protected boolean applicationAlive()
...
+    public void doPrepare()
+    {
+        super.doPrepare();
+        if ( useBundlePluginsIfPresent )
+        {
+            removePluginsPresentInBundle();
+        }
+    }
+
+    private void removePluginsPresentInBundle()
```
Comment (nabcos) [inline nexus/nexus-launcher/src/main/java/org/sonatype/nexus/bundle/launcher/support/DefaultNexusBundle.java:290]: Wrongly named? IIUC this removes plugins in sonatype-work/plugin-repository

Comment (adreghiciu) [inline nexus/nexus-launcher/src/main/java/org/sonatype/nexus/bundle/launcher/support/DefaultNexusBundle.java:290]: No. The switch is to prefer bundle plugins but the way it works is that it lays down then overlays may add plugins (that are added to sonatype-work plugin repository, plugins that are scanned and removed if they exist in bundle

Comment (adreghiciu) [inline nexus/nexus-launcher/src/main/java/org/sonatype/nexus/bundle/launcher/support/DefaultNexusBundle.java:290]: @ben if you can think of a better name let me know

File: nexus/nexus-test-harness/nexus-test-environment-maven-plugin/src/main/java/org/sonatype/plugin/nexus/testenvironment/AbstractEnvironmentMojo.java [hunk 2]
```diff
@@ -1326,4 +1341,87 @@ private void createMarkerFile( String markerName )
...
+                    {
+                        plugins.put( gaCoordinates.get(), plugin );
+                    }
+                }
+            }
+        }
+        return plugins;
+    }
+
+    private Optional<String> getPluginGACoordinates( final File mainJar )
```
Comment (nabcos) [inline nexus/nexus-test-harness/nexus-test-environment-maven-plugin/src/main/java/org/sonatype/plugin/nexus/testenvironment/AbstractEnvironmentMojo.java:1373]: Duplicated code?
https://github.com/sonatype/nexus/pull/582/files#L1R342

File: nexus/nexus-launcher/src/main/java/org/sonatype/nexus/bundle/launcher/support/DefaultNexusBundle.java (hunk 1 above)
Comment (nabcos) [inline nexus/nexus-launcher/src/main/java/org/sonatype/nexus/bundle/launcher/support/DefaultNexusBundle.java:290]: Hm, removeRedundantPluginsInWorkDir? The name to me suggests that this method removes plugins from the bundle, but I see where you're coming from. removePluginsAlreadyPresentInBundle?

Comment (nabcos) [inline nexus/nexus-launcher/src/main/java/org/sonatype/nexus/bundle/launcher/support/DefaultNexusBundle.java:290]: Or maybe just a bit of javadoc to clarify behavior

**Comment by adreghiciu** [comment]
Yes, some code is duplicated as launcher and nexus env plugin they do not have a common ground

**Comment by nabcos** [comment]
+1

File: nexus/nexus-launcher/src/main/java/org/sonatype/nexus/bundle/launcher/support/DefaultNexusBundle.java (hunk 1 above)
Comment (nabcos) [inline nexus/nexus-launcher/src/main/java/org/sonatype/nexus/bundle/launcher/support/DefaultNexusBundle.java:290]: And maybe I should change my HipChat alias to @nabcos as well to not alert someone from github to our discussions =)

**Pull request merged**